opencog/reasoning/RuleEngine/rule-engine-src/pln/PLNCommons.cc

```diff
@@ -51,9 +51,8 @@ Handle PLNCommons::create_quoted(Handle himplicant)
     return hquoted;
 }
 
-Handle PLNCommons::create_bindLink(
-        Handle himplicant, bool vnode_is_typedv)
-                throw (opencog::InvalidParamException)
+Handle PLNCommons::create_bindLink(Handle himplicant, bool vnode_is_typedv)
+    throw (opencog::InvalidParamException)
```

Comment:
FYI,  re throw, I discovered something weird I don't understand:  if I declare a throw, explicitly, like this, then neither gdb nor the stack trace will indicate the correct line number in the C file;  instead, it will use the line number of the very last line of the method.  If I don't declare the throw, then the line numbers work.

There is also a different and much more serious problem: if this method calls another method that throws the same exception, then the exception handling gets completely messed up.  The only way to make the exception work correctly is to remove the  throw declaration. 

I mention this because I've reworked scheme so that scheme exceptions are turned  into c++ exceptions, and c++ exceptinos are turned into scheme exceptions, so that you can throw anywhere for any reason, and can always catch, in either language, and everything "just works".